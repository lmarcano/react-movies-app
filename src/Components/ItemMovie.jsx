import PropTypes from 'prop-types'

//Componente que pinta el Card de cada pelicula
export const ItemMovie = ({movie}) => {

    const imgUrl = 'https://image.tmdb.org/t/p/w200';

    let content;
    if(movie.poster == null){
        content = <figure className="figure">
                    <svg className="bd-placeholder-img figure-img img-fluid" width="400" height="600" xmlns="http://www.w3.org/2000/svg" role="img" aria-label="" preserveAspectRatio="xMidYMid slice" focusable="false"><title>Placeholder</title><rect width="100%" height="100%" fill="#868e96"></rect><text x="50%" y="50%" fill="#dee2e6" dy=".3em"></text></svg>                    
                </figure>
    }else{
        content = <img src={`${imgUrl + movie.poster}`} className="card-img-top" alt={movie.title} />
    }

  return (
    <div className="col-md-3 mb-3 text-center">
        <div className="card shadow" style={{width: 250 + 'px'}}>
            { content }
            <div className="card-body">
                <h5 className="card-title">{movie.title}</h5>                
            </div>
        </div>
    </div>
  )
}

ItemMovie.propTypes = {
    movie: PropTypes.object.isRequired
}