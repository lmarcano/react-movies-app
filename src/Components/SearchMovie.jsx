import { useState } from "react"

import PropTypes from 'prop-types';

//Componente Input Text, que devuelve el texto escrito al padre para realizar una busqueda luego
export const SearchMovie = ({onSearch}) => {

    //variable de estado
    const [inputValue, setInputValue] = useState('');

    //evento del input que retorna lo escrito
    const onInputChange =({target})=>{
        setInputValue(target.value);
    }

    //funcion Submit del formulario
    const onSubmit = (event) => {
        event.preventDefault();

        onSearch(inputValue);
        setInputValue('');
    }    

  return (
    <form className="input-group" onSubmit={onSubmit}>
        
        <input 
            type="text"
            className="form-control"
            data-testid="txSearch"
            placeholder="Escriba el nombre de la pelicula"
            value={inputValue}
            width={40}
            onChange={onInputChange}>
        </input>
        <button onClick={onSubmit} type="button" className="btn btn-primary">Buscar</button>

    </form>
  )
}

//Config de prop types para definir el tipo y hacer obligatorio el envio del parametro onSearch
SearchMovie.propTypes = {
    onSearch: PropTypes.func.isRequired
}