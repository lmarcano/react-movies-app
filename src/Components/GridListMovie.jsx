import { ItemMovie } from "./ItemMovie";

import PropTypes from 'prop-types'

//Componente de la Lista de peliculas
export const GridListMovie = ({movies}) => {    

  return (    
    <>
        <br></br>
        <br></br>
        <h4>POPULARES</h4>
        
        <div className="row">
            {
                movies.map((movie) => (
                    <ItemMovie key={movie.id} movie={movie}></ItemMovie>
                ))
            }
        </div>

    </>
  )
}

GridListMovie.propTypes = {
    movies: PropTypes.array.isRequired
}

//Definir valor de parametros por defecto
GridListMovie.defaultProps = {
    movies: null
}