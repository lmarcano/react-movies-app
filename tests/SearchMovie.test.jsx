import { render, screen } from "@testing-library/react";
import { SearchMovie } from "../src/Components/SearchMovie";

describe('Pruebas een SearchMovie component', () => {

    test('Debe haceer match ccon el snapshot', () => {

        const onSearch = () => {

        }

        const {container} = render(<SearchMovie onSearch={()=>onSearch()}></SearchMovie>);
        
        expect(container).toMatchSnapshot();
    });

    test('Debe exisitir un INPUT', () => {

        render(<SearchMovie onSearch={()=>onSearch()}></SearchMovie>);
        //expect(screen.getByRole('input'));
        
        //screen.debug();

    });

});