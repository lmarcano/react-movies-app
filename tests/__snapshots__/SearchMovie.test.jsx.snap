// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas een SearchMovie component Debe haceer match ccon el snapshot 1`] = `
<div>
  <form
    class="input-group"
  >
    <input
      class="form-control"
      data-testid="txSearch"
      placeholder="Escriba el nombre de la pelicula"
      type="text"
      value=""
      width="40"
    />
    <button
      class="btn btn-primary"
      type="button"
    >
      Buscar
    </button>
  </form>
</div>
`;
